src/libraries/System.Memory/tests/Span/StringSearchValues.cs

```diff
@@ -0,0 +1,450 @@
+                }
+            }
+        }
+
+        [Fact]
+        public static void Create_ThrowsOnNullValues()
+        {
+            Assert.Throws<ArgumentNullException>("values", () => SearchValues.Create(new[] { "foo", null, "bar" }, StringComparison.Ordinal));
+        }
+
+        [Fact]
+        public static void TestIndexOfAny_RandomInputs()
+        {
+            var helper = new StringSearchValuesTestHelper(
+                expected: IndexOfAnyReferenceImpl,
+                searchValues: (searchSpace, values) => searchSpace.IndexOfAny(values));
+
+            helper.TestRandomInputs();
+        }
+
+        [ConditionalFact(nameof(CanTestInvariantCulture))]
+        public static void TestIndexOfAny_RandomInputs_InvariantCulture()
+        {
+            RunUsingInvariantCulture(static () =>
+            {
+                Assert.Equal("Invariant Language (Invariant Country)", CultureInfo.CurrentCulture.NativeName);
+
+                TestIndexOfAny_RandomInputs();
+            });
+        }
+
+        [ConditionalFact(nameof(CanTestNls))]
+        public static void TestIndexOfAny_RandomInputs_Nls()
+        {
+            RunUsingNLS(static () =>
+            {
+                Assert.NotEqual("Invariant Language (Invariant Country)", CultureInfo.CurrentCulture.NativeName);
+
+                TestIndexOfAny_RandomInputs();
+            });
+        }
+
+        [Fact]
+        [ActiveIssue("Manual execution only. Worth running any time SearchValues<string> logic is modified.")]
+        public static void TestIndexOfAny_RandomInputs_Stress()
+        {
+            RunStress();
+
+            if (CanTestInvariantCulture)
+            {
+                RunUsingInvariantCulture(static () => RunStress());
+            }
+
+            if (CanTestNls)
+            {
+                RunUsingNLS(static () => RunStress());
+            }
+
+            static void RunStress()
+            {
+                foreach (int maxNeedleCount in new[] { 2, 8, 20, 100 })
+                {
+                    foreach (int maxNeedleValueLength in new[] { 8, 40 })
+                    {
+                        foreach (int haystackLength in new[] { 100, 1024 })
+                        {
+                            var helper = new StringSearchValuesTestHelper(
+                                expected: IndexOfAnyReferenceImpl,
+                                searchValues: (searchSpace, values) => searchSpace.IndexOfAny(values))
+                            {
+                                MaxNeedleCount = maxNeedleCount,
+                                MaxNeedleValueLength = maxNeedleValueLength,
+                                MaxHaystackLength = haystackLength,
+                                HaystackIterationsPerNeedle = 1_000,
+                            };
+
+                            helper.StressRandomInputs(TimeSpan.FromSeconds(5));
```

Comment:
maybe in this test that's already randomizing, avoid the fixed Random seed?

Reply:
I mean the one in StringSearchValuesTestHelper, like the other one

Reply:
The regular test (`TestIndexOfAny_RandomInputs`) uses a fixed seed to remain [repeatable](https://github.com/dotnet/runtime/pull/76740#discussion_r1012298352) across runs.

This stress helper that just burns CPU to try and find edge cases is using different seeds across runs (note [`rng: new Random()`](https://github.com/dotnet/runtime/blob/112be1b9d055b755e5ee1b27d90c29df8cad7e6e/src/libraries/System.Memory/tests/Span/StringSearchValues.cs#L349)).

Reply:
I meant the stress helper is still relying on this one to set up the initial char arrays.
https://github.com/dotnet/runtime/pull/88394/files/112be1b9d055b755e5ee1b27d90c29df8cad7e6e#diff-6e67f168192eacf1948de7f92fd61db9cc2f084c0c79f3615d8d80eeb0d2fd9bR303